Detectors/TPC/workflow/src/tpc-aggregate-grouped-idc.cxx

```diff
@@ -0,0 +1,99 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include <fmt/format.h>
+#include "Algorithm/RangeTokenizer.h"
+#include "Framework/WorkflowSpec.h"
+#include "Framework/Logger.h"
+#include "Framework/ConfigParamSpec.h"
+#include "Framework/CompletionPolicy.h"
+#include "Framework/CompletionPolicyHelpers.h"
+#include "Headers/DataHeader.h"
+#include "Headers/RAWDataHeader.h"
+#include "CommonUtils/ConfigurableParam.h"
+#include <vector>
+#include <string>
+#include "TPCWorkflow/TPCAggregateGroupedIDCSpec.h"
+#include "TPCCalibration/IDCFactorization.h"
+#include "TPCCalibration/IDCFourierTransform.h"
+
+using namespace o2::framework;
+
+// customize the completion policy
+void customize(std::vector<o2::framework::CompletionPolicy>& policies)
+{
+  using o2::framework::CompletionPolicy;
+  policies.push_back(CompletionPolicyHelpers::defineByName("tpc-idc-aggregate.*", CompletionPolicy::CompletionOp::Consume));
+}
+
+// we need to add workflow options before including Framework/runDataProcessing
+void customize(std::vector<ConfigParamSpec>& workflowOptions)
+{
+  const std::string cruDefault = "0-" + std::to_string(o2::tpc::CRU::MaxCRU - 1);
+
+  std::vector<ConfigParamSpec> options{
+    {"configFile", VariantType::String, "o2tpcaveragegroupidc_configuration.ini", {"configuration file for configurable parameters"}},
+    {"timeframes", VariantType::Int, 10, {"Number of TFs which will be aggregated."}},
+    {"timeframesDeltaIDC", VariantType::Int, 2, {"Number of TFs used for storing DeltaIDC in CCDB."}},
+    {"rangeIDC", VariantType::Int, 10, {"Number 1D-IDCs which will be used for the calculation of the fourier coefficients."}},
+    {"nthreads", VariantType::Int, 1, {"Number of threads which will be used during factorization of the IDCs."}},
+    {"debug", VariantType::Bool, false, {"create debug files"}},
+    {"fft", VariantType::Bool, false, {"using FFT or naive approach"}},
```

Comment:
Shouldn't fft be the default?

Reply:
> Very nice! Only minor things for now. I guess this can go for review now, or are there open things you would like to implement still?

I am currently making some changes to the IDCFourierTransform class. I think after this it can go for review.

Reply:
> Shouldn't fft be the default?

Yes, but when setting `VariantType::Bool` to `true` as a default and then setting it to false from the command line like `o2-tpc-aggregate-idc —fftw false` it will still be recognized as true during the execution. I think it is some bug when using `VariantType::Bool`. There are also some comments in other devices which are I think related to this: https://github.com/AliceO2Group/AliceO2/blob/e4e46e95ee16c3d6b1a276187daccd11cc22899b/Steer/DigitizerWorkflow/src/TOFDigitizerSpec.cxx#L222

Reply:
then perhaps change it to something like `use-naive-fft` and negate the logic.